GVRf/Framework/framework/src/main/java/org/gearvrf/io/GVRInputManager.java

```diff
@@ -212,9 +212,11 @@ private GVRControllerType getGVRInputDeviceType(InputDevice device) {
         int vendorId = device.getVendorId();
         int productId = device.getProductId();
 
-        if (sources == (InputDevice.SOURCE_TOUCHSCREEN | InputDevice.SOURCE_KEYBOARD)) {
+        if ((sources & InputDevice.SOURCE_TOUCHSCREEN) == InputDevice.SOURCE_TOUCHSCREEN
```

Comment:
Isn't it odd that this is the criteria for detecting a controller? A device with touchscreen and keyboard?